From 70b5f00568f14b5db39da7d338f160c35bcf8c62 Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Tue, 9 Mar 2021 12:36:34 -0800
Subject: [PATCH 1763/1852] drm: xlnx: zynqmp_dp: Fix possible crtc searching

commit 097e43b9fe635e61d4f33e1a4ffe2e51d6647f66 from
https://github.com/Xilinx/linux-xlnx.git

Currently the device node of the encoder port is being passed to
drm_of_find_possible_crtcs(). However the function's documentation is
actually incorrect. It requires the entire device OF node, not just the
encoder port OF node.

So go through the entire device OF node. If there is a port, then search
for possible crtcs and break. Else do not search for possible crtcs as
we know there will be none.

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Signed-off-by: Jianqiang Chen <jianqiang.chen@xilinx.com>
Acked-by: Varunkumar Allagadapa <varunkumar.allagadapa@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_dp.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_dp.c b/drivers/gpu/drm/xlnx/zynqmp_dp.c
index 5ced72dc51ee..48cec7aa2a6a 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_dp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_dp.c
@@ -1691,8 +1691,9 @@ int zynqmp_dp_bind(struct device *dev, struct device *master, void *data)
 	for_each_child_of_node(dev->of_node, port) {
 		if (!port->name || of_node_cmp(port->name, "port"))
 			continue;
-		encoder->possible_crtcs |= drm_of_find_possible_crtcs(drm,
-								      port);
+		encoder->possible_crtcs |=
+			drm_of_find_possible_crtcs(drm, dev->of_node);
+		break;
 	}
 	drm_encoder_init(drm, encoder, &zynqmp_dp_encoder_funcs,
 			 DRM_MODE_ENCODER_TMDS, NULL);
-- 
2.31.1

